apps/web/src/components/task/task-comment.tsx

```diff
@@ -36,23 +38,37 @@ function TaskComment() {
     shouldUnregister: true,
   });
 
-  async function onSubmit(data: z.infer<typeof commentSchema>) {
+  async function handleSubmit(data: z.infer<typeof commentSchema>) {
     if (!user?.email) {
       return;
     }
 
     try {
-      await createComment({
-        taskId: taskId,
-        content: data.comment,
-        userEmail: user?.email,
-      });
+      if (commentId) {
+        await updateComment({
+          id: commentId,
+          userEmail: user.email,
+          content: data.comment,
+        });
+        onSubmit?.();
+      } else {
+        await createComment({
+          taskId: taskId,
+          content: data.comment,
+          userEmail: user?.email,
+        });
+        onSubmit?.();
+      }
```

Comment:
Should we verify the user's email beforehand, or let the backend handle it?
If it fails, there's no information except a toast mentioning an internal server error.